webapp/src/vue-apollo.js

```diff
@@ -0,0 +1,78 @@
+import Vue from "vue";
+import VueApollo from "vue-apollo";
+import { createApolloClient, restartWebsockets } from "vue-cli-plugin-apollo/graphql-client";
+
+// Install the vue plugin
+Vue.use(VueApollo);
+
+const AUTH_TOKEN = "apollo-token";
+
+// Http endpoint
+const httpEndpoint = process.env.VUE_APP_GRAPHQL_HTTP || "http://localhost:4000/graphql";
+// Files URL root
+export const filesRoot = process.env.VUE_APP_FILES_ROOT || httpEndpoint.substr(0, httpEndpoint.indexOf("/graphql"));
+
+Vue.prototype.$filesRoot = filesRoot;
+
+// Config
+const defaultOptions = {
+    httpEndpoint,
+};
+
+// Call this in the Vue app file
+export function createProvider(options = {}) {
+    // Create apollo client
+    const { apolloClient, wsClient } = createApolloClient({
+        ...defaultOptions,
+        ...options,
+    });
+    apolloClient.wsClient = wsClient;
+
+    // Create vue apollo provider
+    const apolloProvider = new VueApollo({
+        defaultClient: apolloClient,
+        defaultOptions: {
+            $query: {
+                // fetchPolicy: 'cache-and-network',
+            },
+        },
+        errorHandler(error) {
+            // eslint-disable-next-line no-console
+            console.log(
+                "%cError",
+                "background: red; color: white; padding: 2px 4px; border-radius: 3px; font-weight: bold;",
+                error.message
+            );
+        },
+    });
+
+    return apolloProvider;
+}
+
+// Manually call this when user log in
+export async function onLogin(apolloClient, token) {
```

Comment:
`@nuxtjs/apollo` has useful helpers for this: https://github.com/nuxt-community/apollo-module